### DISCUSSION TITLE
[Feature request] 类似RouterOS 的DNS 缓存功能

### DISCUSSION THREAD

**Discussion by dgulan** [IrineSistiana/mosdns#670] [General]
### 希望添加的功能

以前用的是Openwrt，后来改用RouterOS，遂一直使用RouterOS至今。
RouterOS 有个功能：DNS缓存，就这个功能一直被称为“网页秒开”。
个人主观臆断，使用RouterOS好像是比其它路由器软件打开网页快那么一丝……

想了解下mosdns 是否也有这种dns 缓存功能。

看到issue#454 里面，有看到如下回复：
“unbound 的 预读取 功能设计的非常完美。可控性也很高。有相关需求建议前置一个 unbound。”

只是我搜索了下软件功能介绍，没发现我疑问的答案。
求教了。

谢谢作者的发布，mosdns是一个非常棒的软件。
感谢！

**Comment by chinaniupai** [comment]
> ### 希望增加的功能
> openwrt，routeros，c routeros至今routeros至今routeros有：dns缓存 ， 就打开网页快那么一根……
> 
> 想了想了解下mosdns是否也有这种dns存储功能。
> 
> 看到issue#454里，有看到如下回复： “unbound的预读取功能设计的非常完美。可控制性也很高。有相关需要建议之前设置一个unbound。”
> 
> 只是我搜索了下载软件功能介绍，没有发现我质疑的答案。 求教了。
> 
> 感谢作者的发布，mosdns 是一个非常棒的软件。 谢谢！

win10怎么用呢？

**Comment by kkkgo** [comment]
这里有一个使用了unbound和mosdns的docker，你可以参考一下。
https://github.com/kkkgo/PaoPaoDNS

**Comment by dgulan** [comment]
> > ### 希望增加的功能
> > openwrt，routeros，c routeros至今routeros至今routeros有：dns缓存 ， 就打开网页快那么一根……
> > 想了想了解下mosdns是否也有这种dns存储功能。
> > 看到issue#454里，有看到如下回复： “unbound的预读取功能设计的非常完美。可控制性也很高。有相关需要建议之前设置一个unbound。”
> > 只是我搜索了下载软件功能介绍，没有发现我质疑的答案。 求教了。
> > 感谢作者的发布，mosdns 是一个非常棒的软件。 谢谢！
> 
> win10怎么用呢？

RouterOS，可以看作一个软路由软件。
win10 正常是用不了。

要么你win10 用虚拟机，安装RouterOS来作为路由。

**Comment by dgulan** [comment]
> 这里有一个使用了unbound和mosdns的docker，你可以参考一下。 https://github.com/kkkgo/PaoPaoDNS

谢谢分享。
看了作者的描述，这个递归DNS，我 理解上应该就是RouterOS DNS缓存功能（也有可能我了解错了）

不过这个搭建过程，对于小白稍显复杂，囧。

我开这个帖子，其实就是想了解ModDNS 是否具有这个功能。

我喜欢开源软件。

**Comment by digiw** [comment]
> ### 希望添加的功能
> 以前用的是Openwrt，后来改用RouterOS，遂一直使用RouterOS至今。 RouterOS 有个功能：DNS缓存，就这个功能一直被称为“网页秒开”。 个人主观臆断，使用RouterOS好像是比其它路由器软件打开网页快那么一丝……
> 
> 想了解下mosdns 是否也有这种dns 缓存功能。
> 
> 看到issue#454 里面，有看到如下回复： “unbound 的 预读取 功能设计的非常完美。可控性也很高。有相关需求建议前置一个 unbound。”
> 
> 只是我搜索了下软件功能介绍，没发现我疑问的答案。 求教了。
> 
> 谢谢作者的发布，mosdns是一个非常棒的软件。 感谢！

要是i没记错，routeros好像是闭源的？所以不知道它的缓存有什么绝招，另外说起unbound里的缓存预读取功能，smartdns也有这个选项，不过似乎没觉得性能有加成。 我前几天也尝试想在mosdns v5新版里用redis来做缓存，看看是不是性能有更好提升，但是翻遍文档发现，似乎v5新版取消了redis的直接支持！
暂时没找了，只能先用着v5版本的lazy cache策略，至于redis或者能将解析更进一步的秘技，还得等作者大大来讲解！
